logverification/app/appentry.go

```diff
@@ -208,115 +137,133 @@ func (ae *AppEntry) LogTenant() (string, error) {
-	binary.BigEndian.PutUint64(idTimestampBytes, idTimestamp)
+	copy(mmrSalt[:ExtraBytesSize], extraBytes)
 
-	copy(mmrSalt[ExtraBytesSize:], idTimestampBytes)
+	copy(mmrSalt[ExtraBytesSize:], idTimestamp)
 
 	return mmrSalt, nil
 }
 
-/** Massif gets the massif context, for the massif of the corresponding log entry from the app data.
- *
- * The following massif options can be used, in priority order:
- *   - WithMassifContext
- *   - WithMassifReader
- *   - WithAzblobReader
- *
- * Example WithMassifReader:
- *
- * WithMassifReader(
- *   reader,
- *   WithMassifTenantId("tenant/foo"),
- *   WithMassifHeight(14),
- * )
- */
-func (ae *AppEntry) Massif(options ...MassifGetterOption) (*massifs.MassifContext, error) {
+// MMREntry derives the mmr entry of the corresponding log entry from the app data.
+//
+// MMREntry is:
+//   - H( Domain | MMR Salt | Serialized Bytes)
+//
+// The MMR Salt is sourced from the corresponding log entry
+func (ae *AppEntry) MMREntry(massifContext *massifs.MassifContext) ([]byte, error) {
 
-	massifOptions := ParseMassifGetterOptions(options...)
+	logVersion0 := true
 
-	// first check if the options give a massif context to use, and use that
-	if massifOptions.massifContext != nil {
-		return massifOptions.massifContext, nil
+	extraBytes, err := ae.ExtraBytes(massifContext)
+	if err != nil {
+		return nil, err
 	}
 
-	var massifReader MassifGetter
-	// now check if we have a massif reader
-	if massifOptions.massifGetter != nil {
-		massifReader = massifOptions.massifGetter
-	} else {
-		// otherwise use azblob reader to get it
-		if massifOptions.azblobReader == nil {
-			return nil, errors.New("no way of determining massif of app entry, please provide either a massif context, massif reader or azblob reader")
-		}
-
-		newMassifReader := massifs.NewMassifReader(logger.Sugar, massifOptions.azblobReader)
-		massifReader = &newMassifReader
+	// the only implementation of log version 0 is assetsv2
+	//  so check for the assetsv2 app domain (0).
+	if extraBytes[0] != 0 {
+		logVersion0 = false
 	}
 
-	massifHeight := massifOptions.MassifHeight
+	if logVersion0 {
+		hasher := LogVersion0Hasher{}
 
-	logIdentity := massifOptions.TenantId
-	// if the log identity is not given, attempt to find it from the logId
-	if massifOptions.TenantId == "" {
-		// find the tenant log from the logID
-		logUuid, err := uuid.FromBytes(ae.logID)
+		var idTimestamp []byte
+		idTimestamp, err = ae.IDTimestamp(massifContext)
 		if err != nil {
 			return nil, err
 		}
 
-		// log identity is currently `tenant/logid`
-		logIdentity = fmt.Sprintf("tenant/%s", logUuid.String())
-	}
+		var eventHash []byte
+		eventHash, err = hasher.HashEvent(ae.mmrEntryFields.serializedBytes, idTimestamp)
+		if err != nil {
+			return nil, err
+		}
 
-	return Massif(ae.merkleLogCommit.Index, massifReader, logIdentity, massifHeight)
+		return eventHash, nil
+	}
 
-}
+	// if we get here we know its a log version 1 entry
```

Comment:
Encapsulating the hashing into its own type with `LogVersion0Hasher` seemed quite tidy. Do you feel that's just unnecessary here - perhaps it wouldn't see much re-use?

Reply:
good question!

I think it was a good idea, and good forsight to separate it out.

We should definitely have a logversion1 hasher at some point, but i think its future work TD?

unless you feel its hard to read this method? in which case we can sort it now?